wolfcrypt/src/evp.c

```diff
@@ -6803,6 +6805,874 @@ void wolfSSL_EVP_PKEY_free(WOLFSSL_EVP_PKEY* key)
+        return ASN_PARSE_E;
+
+    if (GetSequence(pkey, &inOutIdx, &length, pkeySz) < 0)
+        return ASN_PARSE_E;
+
+    if (GetASNObjectId(pkey, &inOutIdx, &length, pkeySz) < 0)
+        return ASN_PARSE_E;
+
+    inOutIdx += length;
+
+    if (inOutIdx >= (word32)pkeySz) {
+        return BUFFER_E;
+    }
+
+    localIdx = inOutIdx;
+
+    if (GetASNTag(pkey, &localIdx, &tag, pkeySz) == 0 &&
+        tag == (ASN_SEQUENCE | ASN_CONSTRUCTED)) {
+        return BAD_FUNC_ARG;            /* given key is not a public key*/
+    }
+
+    /* ecc params information */
+    res = GetObjectId(pkey, &inOutIdx, &oidSum, oidIgnoreType, pkeySz);
+    if (res != 0)
+        return res;
+
+    curveId = wc_ecc_get_oid(oidSum, NULL, (word32*)&length);
+    if (curveId < 0 || length == 0) {
+        return ECC_CURVE_OID_E;
+    }
+
+    if (CheckBitString(pkey, &inOutIdx, &length, pkeySz, 1, NULL) < 0) {
+        return ASN_PARSE_E;
+    }
+
+    pointIdx = inOutIdx;
+    pointSz = length;
+    inOutIdx += length;
+
+    nid = EccEnumToNID(curveId);
+
+    /* look up object name from object info table */
+
+    oi = (WOLFSSL_ObjectInfo*)wolfssl_object_info;
+    OIDName = NULL;
+    for (i = 0;i < wolfssl_object_info_sz; i++) {
+        if ( (oi + i)->type == oidCurveType && (oi + i)->nid == nid) {
+            OIDName = (char*)((oi + i)->sName);
+            break;
+        }
+    }
+
+    /* get NIST curve name */
+    nistCurveName = wolfSSL_EC_curve_nid2nist(nid);
+
+    idx = 0;
+    Indent(out, indent);
+
+    if (wolfSSL_BIO_write(out, "Public-Key: (",
+                            (int)XSTRLEN("Public-Key: (")) < 0)
+        return WOLFSSL_FAILURE;
+
+    if (mp_set_int(&a, bitlen) != 0)
+        return WOLFSSL_FAILURE;
+    if (mp_todecimal(&a, (char*)buff) != 0)
+        return WOLFSSL_FAILURE;
+    wsz = (int)XSTRLEN((const char*)buff);
+    if (wolfSSL_BIO_write(out, buff + idx, wsz) <= 0)
+        return WOLFSSL_FAILURE;
+    if (wolfSSL_BIO_write(out, " bit)\n", (int)XSTRLEN(" bit)\n")) <= 0)
+        return WOLFSSL_FAILURE;
+
+    /* print pub element */
+    Indent(out, indent);
+    if (wolfSSL_BIO_write(out, "pub:\n", (int)XSTRLEN("pub:\n")) <= 0)
+        return WOLFSSL_FAILURE;
+    if (DumpElement(out, pkey + pointIdx, pointSz, indent + 4)
+                                                != WOLFSSL_SUCCESS)
+        return WOLFSSL_FAILURE;
+
+    /* print OID in name */
+    Indent(out, indent);
+    if (wolfSSL_BIO_write(out, "ASN1 OID: ", (int)XSTRLEN("ASN1 OID: ")) <= 0)
+        return WOLFSSL_FAILURE;
+
+    if (OIDName && XSTRLEN(OIDName) > 0) {
+        if (wolfSSL_BIO_write(out, OIDName, (int)XSTRLEN(OIDName)) <= 0)
+            return WOLFSSL_FAILURE;
+        if (wolfSSL_BIO_write(out, "\n", 1) < 0)
+            return WOLFSSL_FAILURE;
+    }
+
+    /* print NIST curve name */
+    Indent(out, indent);
+    if (wolfSSL_BIO_write(out, "NIST CURVE: ", (int)XSTRLEN("NIST CURVE: ")) <= 0)
```

Comment:
Wrap lines longer than 80 characters, also applies to line 7256 below.

Reply:
Wrapped those lines.